v4/newrelic/transaction.go

```diff
@@ -49,8 +49,8 @@ func (txn *Transaction) End() {
 	if txn.isEnded() {
 		return
 	}
-	txn.rootSpan.end()
 	txn.thread.Lock()
+	txn.rootSpan.end()
```

Comment:
Now I'm second guessing this change. Is it really necessary to end the span under lock?

Reply:
After an initial look, it seems to me that the locking is important when setting the root span in `setCurrentSpan`, but it does the locking there:

```go
func (thd *thread) setCurrentSpan(s *span) {
	thd.Lock()
	thd.currentSpan = s
	thd.isMultiSpan = true
	thd.Unlock()
}
```
So my first thought is that it shouldn't be necessary, or in fact could maybe be problematic? But I'll think about it some more before I am sure of that. Might be good to write some tests around this if possible.

Reply:
Yeah, I think I'll change this back. Thanks for the sleuthing!